contrib/linux-kernel/lib/zstd/compress.c

```diff
@@ -587,8 +587,10 @@ ZSTD_STATIC size_t ZSTD_compressSequences(ZSTD_CCtx *zc, void *dst, size_t dstCa
 {
 	const int longOffsets = zc->params.cParams.windowLog > STREAM_ACCUMULATOR_MIN;
 	const seqStore_t *seqStorePtr = &(zc->seqStore);
-	U32 count[MaxSeq + 1];
-	S16 norm[MaxSeq + 1];
+	U32 *count = (U32 *)zc->tmpCounters;
+	S16 *norm = (S16 *)(count + MaxSeq + 1);
+	U32 *const workspace = PTR_ALIGN((U32 *)(norm + MaxSeq + 1), sizeof(U32));
```

Comment:
Yeah, wrote this before I saw your second comment. I've fixed it so that it doesn't use the `U16 *` to derive the `U32 *`.